spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -57,6 +75,10 @@ public class BigQueryTemplate implements BigQueryOperations {
 
   private Duration jobPollInterval = Duration.ofSeconds(2);
 
+  private static final int JSON_STREAM_WRITER_BATCH_SIZE = 1000; // write records in batches of 1000
```

Comment:
It would be good to make this configurable by the user -- could you add a property to `GcpBigQueryProperties`, and inject it with a new setter in this class?

Reply:
sure, I have added `spring.cloud.gcp.bigquery.jsonWriterBatchSize` property. Also we are checking if it's set too low (or not set at all), in that case it will be defaulted to `JSON_STREAM_WRITER_BATCH_SIZE`